packages/contract/src/contracts/contract-factory.ts

```diff
@@ -87,18 +85,29 @@ export default class ContractFactory {
 
     const stateRoot = options.stateRoot || getContractStorageRoot(options.storageSlots);
     const contractId = getContractId(this.bytecode, options.salt, stateRoot);
-    const request = new CreateTransactionRequest({
+    const transactionRequest = new CreateTransactionRequest({
       gasPrice: 0,
       gasLimit: MAX_GAS_PER_TX,
       bytecodeWitnessIndex: 0,
       witnesses: [this.bytecode],
       ...options,
     });
-    request.addContractCreatedOutput(contractId, stateRoot);
-    await this.wallet.fund(request);
+    transactionRequest.addContractCreatedOutput(contractId, stateRoot);
+
+    return {
+      contractId,
+      transactionRequest,
+    };
+  }
 
-    const response = await this.wallet.sendTransaction(request);
+  async deployContract(deployContractOptions?: DeployContractOptions) {
+    if (!this.wallet) {
+      return logger.throwArgumentError('Cannot deploy without wallet', 'wallet', this.wallet);
```

Comment:
```suggestion
      return logger.throwArgumentError('Cannot deploy Contract without wallet', this.wallet);
```

Reply:
throwArgumentError requires a message, name and a value. But the message fix makes sense.